Makefile

```diff
@@ -24,14 +25,48 @@ test: ## Runs unit tests
 
 build: ## Build binary
 	@echo Building binary
-	GO111MODULE=on GOOS=linux GOARCH=amd64 CGO_ENABLED=0 $(GO) build -gcflags all=-trimpath=$(PWD) -asmflags all=-trimpath=$(PWD) -a -installsuffix cgo -o build/_output/bin/chimera -ldflags '$(LDFLAGS)' ./cmd/chimera
+	@if [ "$(ARCH)" = "amd64" ]; then \
+		export GOARCH="amd64"; \
+	elif [ "$(ARCH)" = "arm64" ]; then \
+		export GOARCH="arm64"; \
+	elif [ "$(ARCH)" = "arm" ]; then \
+		export GOARCH="arm"; \
+	else \
+		echo "Unknown architecture $(ARCH)"; \
+		exit 1; \
+	fi; \
+	GO111MODULE=on GOOS=linux CGO_ENABLED=0 $(GO) build -gcflags all=-trimpath=$(PWD) -asmflags all=-trimpath=$(PWD) -a -installsuffix cgo -o build/_output/bin/chimera -ldflags "$(LDFLAGS)" ./cmd/chimera
 
 build-image: ## Build Chimera docker image
 	@echo Building Docker image
-	docker build . -t $(IMAGE) \
-	--build-arg DOCKER_BUILDER_IMAGE=$(DOCKER_BUILDER_IMAGE) \
+	@if [ -z "$(DOCKER_USERNAME)" ] || [ -z "$(DOCKER_PASSWORD)" ]; then \
+		echo "DOCKER_USERNAME and/or DOCKER_PASSWORD not set. Skipping Docker login."; \
+	else \
+		echo $(DOCKER_PASSWORD) | docker login --username $(DOCKER_USERNAME) --password-stdin; \
+	fi
+	docker buildx build \
+	--platform linux/arm64,linux/amd64 \
+	--build-arg DOCKER_BUILD_IMAGE=$(DOCKER_BUILDER_IMAGE) \
+	--build-arg DOCKER_BASE_IMAGE=$(DOCKER_BASE_IMAGE) \
+	. -f Dockerfile -t $(IMAGE) \
+	--no-cache \
+	--push
```

Comment:
Hu, it's a bit surprising that a `build-image` target also pushes the image. (At least, that is what I get from https://docs.docker.com/engine/reference/commandline/buildx_build/#image) Can we make the target name more clear or split it into separate commands?

Reply:
Let me give some reasoning around this. When the build supports multiple architectures since usually the place we are going to test it is not our local machine but cloud environment we prefer to push at the same build step. I understand that might be confusing initally but it is our standard process in all mattermost cloud tooling.
If you need a build for local testing feel free to add a specific block on the makefile that serves this purpose.

Reply:
If that is the SOP for Mattermost Cloud tools, I'm happy to follow it :+1: